app/models/institutions_archive.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  # rubocop:disable Metrics/MethodLength
+  def self.archive_previous_versions(user)
```

Comment:
If I'm reading this right, this only archives and then deletes the most recent previous version.
How are you planning to archive all the existing previous versions in the table?

Reply:
Correct, it will only archive previous production version and preview versions less than the newest production version.  This was done to reduce the database impact.  The plan would then be to manually archive all the older stuff so that you wouldn't have a 5 million insert/delete on the first run of the archive process.

Reply:
We have been working on this PR to get the table to hold only the latest version and move all the historical data to an archive table. We are open to suggestions for the best way to achieve this goal without causing any production performance issues. We can setup a meeting, if that would be beneficial.

Reply:
No, I understand.  I suggested something similar previous.  I will suggest changing the name of the method to `self.archive_previous_version` (singular)